src/main/java/games/strategy/ui/SwingComponents.java

```diff
@@ -324,34 +323,38 @@ public static JMenu newJMenu(final String menuTitle, final KeyboardCode keyboard
     return menu;
   }
 
+  public enum FolderSelectionMode {
+    FILES, DIRECTORIES
+  }
+
+  public static Optional<File> showJFileChooserForFiles() {
+    return showJFileChooser(FolderSelectionMode.FILES);
+  }
+
+  public static Optional<File> showJFileChooserForFolders() {
+    return showJFileChooser(FolderSelectionMode.DIRECTORIES);
+  }
+
   /**
-   * Creates a new component that emulates a multiline label.
-   *
-   * <p>
-   * The multiline label will properly wrap text that has embedded newlines ({@code \n}).
-   * </p>
-   *
-   * @param text The text to be displayed; may be {@code null}.
-   * @param rows The number of rows; must be greater than or equal to zero.
-   * @param cols The number of columns; must be greater than or equal to zero.
-   *
-   * @return The new multiline label; never {@code null}.
-   *
-   * @throws IllegalArgumentException If {@code rows} or {@code cols} is negative.
+   * Shows a dialog the user can use to select a folder or file.
+   * @param folderSelectionMode Flag controlling whether files or folders are available for selection.
+   * @return Empty if the user selects nothing, otherwise the users selection.
    */
-  public static JTextArea newMultilineLabel(final String text, final int rows, final int cols) {
-    checkArgument(rows >= 0, "rows must not be negative");
-    checkArgument(cols >= 0, "cols must not be negative");
+  public static Optional<File> showJFileChooser(final FolderSelectionMode folderSelectionMode) {
+    final JFileChooser fileChooser = new JFileChooser();
+    if (folderSelectionMode == FolderSelectionMode.DIRECTORIES) {
+      fileChooser.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+    } else {
+      fileChooser.setFileSelectionMode(JFileChooser.FILES_ONLY);
+    }
+
+    final int result = fileChooser.showOpenDialog(null);
```

Comment:
Similar issue reported above about not having a proper parent frame.  There's already a few file chooser methods in `GameRunner`.  I don't like to keep throwing UI stuff over there, but that seems to be the short-term solution we've chosen to address this problem...